Navtor_intern_task/FleetManagementApp/Ship.cs

```diff
@@ -0,0 +1,96 @@
+using System.Text.RegularExpressions;
+
+namespace FleetManagementApp;
+
+public abstract partial class Ship
+{
+    public string Id { get; private set; }
+    public string Name { get;}
+    public int Width { get;} // in meters
+    public int Length { get;} // in meters
+
+    public LinkedList<Position> Positions = [];
+    
+    
+
+    protected Ship(string id, string name, int width, int length, Position currentPosition)
```

Comment:
Missing validation for other properties. Negative `Width`? What are possible coordinate values?